compiler/front/msgs.nim

```diff
@@ -654,6 +664,35 @@ template globalReport*(conf: ConfigRef, report: ReportTypes) =
   handleReport(
     conf, wrap(report, instLoc()), instLoc(), doRaise)
 
+proc reportAndFail*(
+  conf: ConfigRef, r: Report, reportFrom: InstantiationInfo) =
+  ## Similar to `handleReport`, but, unless overridden with aborting
+  ## (`doAbort`) by the structured report hook, always raises a recoverable
+  ## error.
+  var rep = r
+  fillReportAndHandleVmTrace(conf, rep, reportFrom)
+
+  let userAction = conf.report(rep)
+  case userAction
+  of doAbort:
+    quit 1
+  of doDefault:
+    let (action, trace) = errorActions(conf, rep, userAction)
+    case action
+    of doAbort:
+      quit 1
```

Comment:
```suggestion
  case conf.report(rep)
  of doAbort:
    quit(conf, false)
  of doDefault:
    let (action, trace) = errorActions(conf, rep, doDefault)
    case action
    of doAbort:
      quit(conf, trace)
```
Two things:
- the compiler's `quit` overload needs to be used
- the value passed to `errorActions` is always `doDefault`, so the `userAction` local is unnecessary